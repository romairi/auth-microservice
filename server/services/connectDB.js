import mongoose from "mongoose";
import {serverConfig} from "../config/serverConfig";
import Logger from "./logger";

/**
 * Connection to MongoDB with specifying several more parameters in the URI
 */
export const connectDB = (logger = new Logger(), {process: _process} = globalThis) => {
    return mongoose.connect(serverConfig.mongo.hostURI, {
        useNewUrlParser: true,
        useUnifiedTopology: true,
        useFindAndModify: false,
        useCreateIndex: true
    })
        .then(() => logger.log('MongoDB Connected...'))
        .catch(err => {
            logger.log("MONGO ERROR");
            logger.error(err.message);
            _process.exit(1);
        })
};
