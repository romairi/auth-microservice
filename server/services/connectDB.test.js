import mongoose from "mongoose";

import { connectDB } from "./connectDB";

jest.mock("mongoose");
jest.mock("../config/serverConfig", () => ({
  serverConfig: {
    mongo: {
      hostURI: "MOCK_URI",
    },
  },
}));

let mockLogger = {
  log: jest.fn(),
  error: jest.fn(),
};

let mockProcess = {
  exit: jest.fn(),
};

describe("connectDB", () => {
  beforeEach(() => {
    mockLogger.log.mockClear();
    mockLogger.error.mockClear();
    mockProcess.exit.mockClear();
  });

  it("should successfully connect to mongoDB", async () => {
    mongoose.connect.mockImplementationOnce(() => Promise.resolve());

    await connectDB(mockLogger, { process: mockProcess });

    expect(mongoose.connect.mock.calls.pop()).toMatchSnapshot();
    expect(mockLogger.log.mock.calls.pop()).toMatchSnapshot();
    expect(mockLogger.error).not.toHaveBeenCalled();
    expect(mockProcess.exit).not.toHaveBeenCalled();
  });

  it("should failed to connect to mongoDB", async () => {
    mongoose.connect.mockImplementationOnce(() =>
      Promise.reject(new Error("ERROR_MESSAGE"))
    );

    await connectDB(mockLogger, { process: mockProcess });

    expect(mongoose.connect.mock.calls.pop()).toMatchSnapshot();
    expect(mockLogger.log.mock.calls.pop()).toMatchSnapshot();
    expect(mockLogger.error.mock.calls.pop()).toMatchSnapshot();
    expect(mockProcess.exit).toHaveBeenCalledWith(1);
  });
});
