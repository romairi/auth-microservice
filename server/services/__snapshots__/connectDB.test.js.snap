// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectDB should failed to connect to mongoDB 1`] = `
Array [
  "MOCK_URI",
  Object {
    "useCreateIndex": true,
    "useFindAndModify": false,
    "useNewUrlParser": true,
    "useUnifiedTopology": true,
  },
]
`;

exports[`connectDB should failed to connect to mongoDB 2`] = `
Array [
  "MONGO ERROR",
]
`;

exports[`connectDB should failed to connect to mongoDB 3`] = `
Array [
  "ERROR_MESSAGE",
]
`;

exports[`connectDB should successfully connect to mongoDB 1`] = `
Array [
  "MOCK_URI",
  Object {
    "useCreateIndex": true,
    "useFindAndModify": false,
    "useNewUrlParser": true,
    "useUnifiedTopology": true,
  },
]
`;

exports[`connectDB should successfully connect to mongoDB 2`] = `
Array [
  "MongoDB Connected...",
]
`;
